includes/utilities.inc

```diff
@@ -47,6 +49,8 @@ function islandora_rest_get_response($endpoint, array $path_parameters) {
     $response = array('message' => $e->getMessage());
   }
   $encode = is_array($response) || is_object($response);
+  $http_response_code = (isset($e) && method_exists($e, 'getCode') ? $e->getCode() : '200');
+  module_invoke_all('islandora_rest_response', $http_response_code, $endpoint, $request_method, $pid, $dsid, $parameters['request']);
```

Comment:
@mjordan I'm taking a look at this and it seems like this hook isn't generic enough.  I think it should mimic the current function's signature plus the return from the response function and the exception object.  This function has been designed to be agnostic as to what it is used for and the current design of the hook bakes in some assumptions that the current function's signature does not. An alternative approach could be to implement the request tracking via menu_alter as islandora_usage_stats does, but passing through the current route rather than hard wiring would be required if there were to be more than one alter. If you'd like to chat about this we can arrange a call.